code/modules/mob/living/living.dm

```diff
@@ -830,9 +830,16 @@
 /mob/living/proc/can_use_vents()
 	return "You can't fit into that vent."
 
+//Checks for anything other than eye protection that would stop flashing. Overridden in carbon.dm and human.dm
+/mob/living/proc/can_be_flashed()
+	if(HAS_TRAIT(src, TRAIT_BLIND) || HAS_TRAIT(src, TRAIT_FLASH_PROTECTION))
+		return FALSE
+
+	return TRUE
+
 //called when the mob receives a bright flash
 /mob/living/proc/flash_eyes(intensity = 1, override_blindness_check = 0, affect_silicon = 0, visual = 0, laser_pointer = FALSE, type = /obj/screen/fullscreen/flash)
-	if(check_eye_prot() < intensity && (override_blindness_check || !HAS_TRAIT(src, TRAIT_BLIND)) && !HAS_TRAIT(src, TRAIT_FLASH_PROTECTION))
+	if(check_eye_prot() < intensity && (override_blindness_check || can_be_flashed()))
```

Comment:
I'd move this whole check to the `can_be_flashed`. You'll need to pass the `intensity` and `override_blindness_check` parameters as well for that. But then the logic will stay the same.
Now `override_blindness_check` will also override you not having eyes etc.